### ISSUE TITLE
Confusion on calculating residual sum squares in modkmeans

### ISSUE THREAD

**Issue by neurothew** [wmvanvliet/mne_microstates#2]
Hello Marijn,

I have come across with the Poulsen's paper on the eeglab microstate toolbox and attempted to write my very own version to facilitate my learning on maths behind. Then I noticed you have written it for MNE, which is really great!

My question isnt really about the code itself but the maths behind it. In the modified k-means algorithm, if I understood correctly, the following two lines calculate the present data sum of squares (within that iteration) and subtracted it from the original data sum of squares.
`
act_sum_sq = np.sum(np.sum(maps[segmentation].T * data, axis=0) ** 2)
`
`residual = abs(data_sum_sq - act_sum_sq)`

In the original paper by Pascual-Marqini (1995), this step is described in Table I, step 5 as:
![image](https://user-images.githubusercontent.com/24700304/68292895-cb729f00-00c7-11ea-9afd-3081cc6d5d11.png)

The major part of the equation consists of subtraction of two dot products, one is the dot product between data vector, another is the dot product of maps vector and data vector, so in this case it can be regarded as element-wise matrix multiplication between maps and data.

I have checked both in Poulsen's eeglab toolbox and your code but I could not understand why we have to sum the product of maps and data along axis 0? Shouldnt it be just like
`np.sum((maps[segmentation].T * data)**2)`
just like what we calculate the data_sum_sq as
`np.sum(data**2)` ?

Any help is much appreciated, thank you very much!

**Comment by wmvanvliet** [comment]
I think you just found a bug in the code. I fixed it in current master. Could you give it a try and see if it works ok for you?

**Comment by neurothew** [comment]
> I think you just found a bug in the code. I fixed it in current master. Could you give it a try and see if it works ok for you?

I actually tried it myself before I asked, but it was incorrect because such calculations would severely lower the explained variance (as calculated in Poulsen's EEGlab toolbox, but not implemented in your script).

Then, I tried to follow the original equation by calculating the sum of squares of dot product between maps and data vector and achieved the same results of their implementation in EEGlab toolbox. 

Suppose i define map vector as a_n and data as x_n, the crucial point to use sum of elementwise square like `np.sum(data**2)` is that we have to create a matrix which columns are a_n^T * x_n, but that was not equivalent to `maps[segmentation].T * data`, which is why i made the mistake.

So thanks for the help and i suggest you to revert the changes!
